lib/client.go

```
@@ -651,6 +665,15 @@ func (c *Client) SetOptions(env string, timeout int) error {
 
 	c.Options.Timeout = &timeout
 
+	client := c.Options.HTTPClient
+	if client == nil {
+		client = &http.Client{
+			Timeout: time.Second * time.Duration(timeout),
```

Took me a bit to wrap my head around the control flow here. I might throw an error if both `c.Options.HTTPClient` is set, and `timeout` is non-zero, because otherwise we'd be silently ignoring an option.

---

Done

---

Actually I think we need to do this in the constructor as well.